eox_nelp/pearson_vue/rti_backend.py

```diff
@@ -0,0 +1,49 @@
+"""
+This module provides the RealTimeImport class, which is responsible for orchestrating the RTI pipeline
+and executing various processes related to rti.
+
+Classes:
+    RealTimeImport: Class for managing RTI operations and executing the pipeline.
+"""
+
+
+class RealTimeImport:
+    """
+    Class for managing RTI (Real Time Import) operations and executing the pipeline.
+
+    Attributes:
+        backend_data (dict): A dictionary containing backend-specific data.
+
+    Methods:
+        run_pipeline(): Executes the RTI pipeline by iterating through the pipeline functions.
+        get_pipeline(): Returns the RTI pipeline, which is a list of functions to be executed.
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Initializes the RealTimeImport instance with the provided keyword arguments.
+
+        Args:
+            **kwargs: Additional keyword arguments to configure the RealTimeImport instance.
+        """
+        self.backend_data = kwargs.copy()
+
+    def run_pipeline(self):
+        """
+        Executes the RTI pipeline by iterating through the pipeline functions.
+        """
+        pipeline = self.get_pipeline()
+        pipeline_index = self.backend_data.get("pipeline_index", 0)
+
+        for idx, func in enumerate(pipeline[pipeline_index:]):
+            self.backend_data["pipeline_index"] = pipeline_index + idx
+            result = func(**self.backend_data) or {}
```

Comment:
```suggestion
            result = func(**self.backend_data) or {}
            self.backend_data["pipeline_index"] = pipeline_index + idx
```
may be the pipeline_index would be updated after the function works, if failsI  think that the pipeline would retry the function that fails. and  Not the next one  🤔

Reply:
Yes, that is the idea if something fails that should resume from the failure pipeline not the next one

Reply:
I validate with the last test and kwnowin that idx starts in zero, the position is ok to recall that position.